Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1331,6 +1335,17 @@ class BrowserViewController: UIViewController {
                 // for reasons above about URL spoofing risk.
                 navigateInTab(tab: tab)
             }
+            
+            // Rewards reporting
+            if let url = change?[.newKey] as? URL, !url.isLocal {
```

Comment:
`tab.url` was often reporting previous url instead of new value, changed to use the kvo's `newKey`